Sources/NIOPosix/BSDSocketAPICommon.swift

```diff
@@ -127,6 +127,28 @@ extension NIOBSDSocket.Option {
         NIOBSDSocket.Option(rawValue: Posix.IPV6_RECVPKTINFO)
 }
 
+extension NIOBSDSocket {
+    struct ProtocolSubtype: RawRepresentable, Hashable {
+        typealias RawValue = CInt
+        var rawValue: RawValue
+        
+        init(rawValue: RawValue) {
+            self.rawValue = rawValue
+        }
+    }
+}
+
+extension NIOBSDSocket.ProtocolSubtype {
+    static let ip = Self(rawValue: 0)
```

Comment:
I don't think `0` implies `IP`: I think it just implies "default".

Reply:
Done in dddf452e0e9843312528554df56be8d8cbe0fbd2